web/relative.go

```diff
@@ -17,6 +18,17 @@ func RelativeEndpoint(request *http.Request, endpoint string) string {
 		proto = "https"
 	}
 
+	// If a client sends a HTTP Host header that includes the default port
+	// for a scheme then we strip the port out of the host and return the
+	// standards compliant host instead. This is mainly done to prevent
+	// returning a directory to the user that includes the port, which they
+	// would then use in the 'url' JWS signature header.
+	if proto == "https" && strings.HasSuffix(host, ":443") {
```

Comment:
Can you break out this repeated chunk of code into a helper function that can be used both places? And that way we can explain in the comment why they are linked - if we always give out directory URLs that don't have a `:443`, then we should always expected the `"url"` field in JWS to not have a `:443`